python.md

```diff
@@ -0,0 +1,216 @@
+  names](https://duckduckgo.com/?q=There+are+only+two+hard+things+in+Computer+Science%3A+cache+invalidation+and+naming+things)
+  right away. :P
+
+- **Special case:** Also align continued lines of continued lines:
+
+  ```python
+
+  # YES:
+  raise Exception("Lorem {0} ipsum {1} dolor sit amet, {2} consectetur {3} ad "
+                  "{4} rcitatio {5}.".format(reprehenderit, voluptate, velit,
+                                             pariatur, deserunt, laborum))
+  ```
+
+### Code Documentation
+*[cf. Google Python Style Guide [Comments and
+Docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)]*
+
+#### General Recommendations
+
+- Write docstrings and comments before or as you write your code. You
+  will never again understand your code better.
+- Don't state the obvious! Don't spam! If someone else (including the future
+  you) thinks it is less effort to decipher your code than to read its
+  documentation, then you probably commented too much. Plus, more docs means
+  more maintenance work.
+- Code documentation that contradicts the code is worse than no code
+  documentation. Always make a priority of keeping the docstrings and comments
+  up-to-date when the code changes.
+
+#### Docstrings
+
+- Add at least the docstring sections described in the [Google Python Style
+  Guide](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).
+- Add a lab-specific `Side Effects` section, for any side
+  effects of your function. If you have doubts about whether your function has
+  side effects, consult with the reviewer.
+- Add other sections supported by the [Napoleon Sphinx
+  extension](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/#docstring-sections)
+  as you see fit.
+- `Returns` should be the last section.
+- Don't add empty sections.
+- Generally try to minimize vertical space by avoiding redundancy and
+  aiming for conciseness. It is desirable to fit docstring and function
+  body on a reasonably sized screen.
+- Use vanilla
+  [reStructuredText/Sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html)
+  markup sparingly in order to enhance visual appeal on [*Read the
+  Docs*](https://docs.readthedocs.io/en/stable/), but without impeding
+  readability of raw docstrings. For instance, it is a good idea to mark up a
+  block of code using the rather subtle [double colon, blank line and
+  indentation](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks),
+  but not to litter a docstring with [Sphinx
+  directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html).
+
+
+#### TODO-comments
+*[cf. Google Python Style Guide [TODO Comments](
+https://google.github.io/styleguide/pyguide.html#312-todo-comments)]*
+
+Adding a parenthesized identifier in a `# TODO:`-comment is not necessary. The
+version control system can provide that information just as well.
+
+### Strings
```

Comment:
IMO we should add a clear recommendation for either double or single quoted strings here. See https://github.com/secure-systems-lab/code-style-guidelines/issues/4#issuecomment-729696991.

Reply:
Any preferences, anyone?

Reply:
I like double quotes because of the prevalence of apostrophes, but either is fine.

Reply:
I default to double quotes as well.

Reply:
Thanks! I added a recommendation in 6837167.